ssl/statem/statem_srvr.c

```diff
@@ -1263,6 +1449,15 @@ MSG_PROCESS_RETURN tls_process_client_hello(SSL *s, PACKET *pkt)
         goto err;
     }
 
+    /* Check we've got a key_share for TLSv1.3 */
+    if (s->version == TLS1_3_VERSION && s->s3->peer_tmp == NULL && !s->hit) {
```

Comment:
Then we could even have clear documentation that it means the session was resumed, whether by ticket or session cache [or other means?].